[FIX] LiveChat switch department not working

renatobecker opened the pull request:
Closes #7397 

When a livechat visitor attempted to change the current department, they were staying in the same department.
This was happening because the new department data was not being updated in the visitor collection.

<img width="138" alt="screen shot 2018-06-05 at 16 30 40" src="https://user-images.githubusercontent.com/2067649/40998321-d456a6b2-68dd-11e8-8fed-3b659786b0b7.png">

renatobecker commented:
@ggazzo:

Related to your first question, that behaviour has been fixed here:https://github.com/RocketChat/Rocket.Chat/pull/10695/files#diff-e531651dd1b362372762eb86f21419f0R76

About the second question, I changed the method called in the Livechat widget. I saw that when an agent was calling the switch department on their side, the behaviour of the process was different from when it was called by Livechat's side, but I do not know the reason for that, so I unified the method called in both sides. Now, when a visitor tries to switch departments, an agent is automatically assigned to the chat (by using `Least Amount` routing). In my opinion, this behaviour will provide a better experience for visitors.
Also, I've improved a few other things, such as when there are no agents available in the department selected by the visitor, in this case a warning will be displayed to the visitor.

I'll be waiting for your review.
Thanks.

ggazzo commented on packages/rocketchat-livechat/.app/client/views/switchDepartment.js:
```diff
@@ -36,40 +36,36 @@ Template.switchDepartment.events({
-			if (!result.value) {
-				return;
-			}	
+		}, () => {
```
I think we updated the sweetalert and now its promise based, we should change to 
```javascript
}).then(() => {
```

renatobecker commented:
@ggazzo, I just fixed this PR.
Yes, we have changed Livechat's `swal` a few weeks ago, but due to the asynchronous review process, change is needed here, but now everything is fine.
The other case, related to displaying the new `agent` after changing the department, is also fixed.
The bug was happening because we changed the object passed through `RocketChat.Livechat.stream`, so because of that the stream was no longer being emitted.

But now everything is OK!

Thanks.
